diff --git a/packages/glib/build.sh b/packages/glib/build.sh
index faa65c22f..787414ee7 100644
--- a/packages/glib/build.sh
+++ b/packages/glib/build.sh
@@ -8,12 +8,10 @@ TERMUX_PKG_SRCURL=https://download.gnome.org/sources/glib/${TERMUX_PKG_VERSION%.
 TERMUX_PKG_SHA256=9f23a9de803c695bbfde7e37d6626b18b9a83869689dd79019bf3ae66c3e6771
 TERMUX_PKG_AUTO_UPDATE=true
 TERMUX_PKG_DEPENDS="libffi, pcre2, resolv-conf, zlib"
-TERMUX_PKG_BUILD_DEPENDS="gobject-introspection"
 TERMUX_PKG_BREAKS="glib-dev"
 TERMUX_PKG_REPLACES="glib-dev"
 TERMUX_PKG_DISABLE_GIR=false
 TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
--Dintrospection=enabled
 -Druntime_dir=$TERMUX_PREFIX/var/run
 -Dlibmount=disabled
 -Dman-pages=enabled
@@ -70,8 +68,12 @@ termux_step_pre_configure() {
 
 	TERMUX_PKG_VERSION=. termux_setup_gir
 
-	# Workaround: Remove cyclic dependency between gir and glib
-	sed -i "/Requires:/d" "${TERMUX_PREFIX}/lib/pkgconfig/gobject-introspection-1.0.pc"
+	# Workaround: cyclic dependency between gir and glib
+	if [ -f "${TERMUX_PREFIX}/lib/pkgconfig/gobject-introspection-1.0.pc" ]; then
+		sed -i "/Requires:/d" "${TERMUX_PREFIX}/lib/pkgconfig/gobject-introspection-1.0.pc"
+	else
+		TERMUX_PKG_EXTRA_CONFIGURE_ARGS+=" -Dintrospection=disabled"
+	fi
 }
 
 termux_step_post_make_install() {
@@ -84,11 +86,15 @@ termux_step_post_make_install() {
 	done
 
 	# Workaround: Restore deleted line in pre-configure step
-	echo "Requires: glib-2.0 gobject-2.0" >> "${TERMUX_PREFIX}/lib/pkgconfig/gobject-introspection-1.0.pc"
+	if [ -f "${TERMUX_PREFIX}/lib/pkgconfig/gobject-introspection-1.0.pc" ]; then
+		echo "Requires: glib-2.0 gobject-2.0" >> "${TERMUX_PREFIX}/lib/pkgconfig/gobject-introspection-1.0.pc"
+	fi
 }
 
 termux_step_post_massage() {
-	rm -v lib/pkgconfig/gobject-introspection-1.0.pc
+	if [ -f "lib/pkgconfig/gobject-introspection-1.0.pc" ]; then
+		rm -v lib/pkgconfig/gobject-introspection-1.0.pc
+	fi
 }
 
 termux_step_create_debscripts() {
diff --git a/scripts/build/termux_step_get_dependencies.sh b/scripts/build/termux_step_get_dependencies.sh
index f1344c93a..876b010f7 100644
--- a/scripts/build/termux_step_get_dependencies.sh
+++ b/scripts/build/termux_step_get_dependencies.sh
@@ -46,7 +46,18 @@ termux_step_get_dependencies() {
 
 		if $build_dependency; then
 			DEB_FILE_TO_INSTALL=$TERMUX_OUTPUT_DIR/${PKG}_${DEP_VERSION}_${DEP_ARCH}.deb
-			if [ -f "$DEB_FILE_TO_INSTALL" ]; then
+			GLIB_PKG="glib"
+			GIR_PKG="gobject-introspection"
+			GLIB_GIR_CYCLE_MARK="$TERMUX_SCRIPTDIR/$GLIB_PKG-$GIR_PKG-cycle-begun"
+			if [ "$PKG" = "$GLIB_PKG" ] && [ ! -f "$GLIB_GIR_CYCLE_MARK" ]; then
+				touch "$GLIB_GIR_CYCLE_MARK"
+				termux_run_build-package # glib 1
+				PKG_DIR="$GIR_PKG"
+				termux_run_build-package # gir 1
+				PKG_DIR="$GLIB_PKG"
+				termux_run_build-package # glib 2
+				rm -r "$TERMUX_TOPDIR/$GIR_PKG/massage/"* # for gir 2 later
+			elif [ -f "$DEB_FILE_TO_INSTALL" ]; then
 				echo "Using already built $DEB_FILE_TO_INSTALL ..."
 			else
 				[ ! "$TERMUX_QUIET_BUILD" = true ] && echo "Building dependency $PKG instead of downloading..."
