--- a/termux-widget/app/build.gradle
+++ b/termux-widget/app/build.gradle
@@ -45,6 +45,8 @@ android {
     }
 
     compileOptions {
+        coreLibraryDesugaringEnabled true
+
         sourceCompatibility JavaVersion.VERSION_11
         targetCompatibility JavaVersion.VERSION_11
     }
@@ -63,6 +65,7 @@ android {
 }
 
 dependencies {
+    coreLibraryDesugaring "com.android.tools:desugar_jdk_libs:2.1.2"
     // required for TermuxWidgetControlsProviderService
     implementation "org.reactivestreams:reactive-streams:1.0.3"
     implementation "io.reactivex.rxjava2:rxjava:2.2.10"
