--- a/packages/apt/build.sh
+++ b/packages/apt/build.sh
@@ -3,8 +3,8 @@ TERMUX_PKG_DESCRIPTION="Front-end for the dpkg package manager"
 TERMUX_PKG_LICENSE="GPL-2.0"
 TERMUX_PKG_MAINTAINER="@termux"
 TERMUX_PKG_VERSION="3.1.15"
-TERMUX_PKG_SRCURL=https://deb.debian.org/debian/pool/main/a/apt/apt_${TERMUX_PKG_VERSION}.tar.xz
-TERMUX_PKG_SHA256=208d58ff5bf6d9ebb93da0d77d480ef87b4618b0210dc88cb977aadd88d1c1c6
+TERMUX_PKG_SRCURL="https://salsa.debian.org/apt-team/apt/-/archive/$TERMUX_PKG_VERSION/apt-$TERMUX_PKG_VERSION.tar.gz"
+TERMUX_PKG_SHA256=41f1d0c3ffca5bb4a2beda0a015cd5c1a9e5324df42da4151761b4ac388793e1
 # apt-key requires utilities from coreutils, findutils, gpgv, grep, sed.
 TERMUX_PKG_DEPENDS="coreutils, dpkg, findutils, gpgv, grep, libc++, openssl, liblzma, sed, termux-keyring, termux-licenses, xxhash"
 TERMUX_PKG_BUILD_DEPENDS="docbook-xsl"
