--- a/packages/debianutils/build.sh
+++ b/packages/debianutils/build.sh
@@ -3,8 +3,8 @@ TERMUX_PKG_DESCRIPTION="Small utilities which are used primarily by the installa
 TERMUX_PKG_LICENSE="GPL-2.0"
 TERMUX_PKG_MAINTAINER="@termux"
 TERMUX_PKG_VERSION="5.23.1"
-TERMUX_PKG_SRCURL=https://deb.debian.org/debian/pool/main/d/debianutils/debianutils_${TERMUX_PKG_VERSION}.tar.xz
-TERMUX_PKG_SHA256=206c669cbf431da30904d4f9e69d049cb711714f5c137b66bf0b1f66d58710bc
+TERMUX_PKG_SRCURL=https://salsa.debian.org/debian/debianutils/-/archive/debian/${TERMUX_PKG_VERSION}/debianutils-debian-${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=cd1a24200d56efe515a8baefc8579a5fa1b2442dc3868c945bf54533a4aca7d6
 TERMUX_PKG_AUTO_UPDATE=true
 
 TERMUX_PKG_RM_AFTER_INSTALL="
