--- a/packages/glib/build.sh
+++ b/packages/glib/build.sh
@@ -65,6 +65,12 @@ termux_step_host_build() {
 }
 
 termux_step_pre_configure() {
+	# Remove this marker all the time in Google Play Termux
+	# because this termux_step_host_build() installs files into $TERMUX_PREFIX (glib-cross),
+	# but Google Play Termux's build-package.sh deletes the entire $TERMUX_PREFIX in between
+	# building every package, so it isn't designed to "understand" this kind of
+	# termux_step_host_build().
+	rm -rf $TERMUX_HOSTBUILD_MARKER
 	# glib checks for __BIONIC__ instead of __ANDROID__:
 	CFLAGS+=" -D__BIONIC__=1"
 	_PREFIX="$TERMUX_PKG_TMPDIR/prefix"
