diff --git a/packages/libxml2/build.sh b/packages/libxml2/build.sh
index 98e1a70..d1106f3 100644
--- a/packages/libxml2/build.sh
+++ b/packages/libxml2/build.sh
@@ -2,9 +2,9 @@ TERMUX_PKG_HOMEPAGE=http://www.xmlsoft.org
 TERMUX_PKG_DESCRIPTION="Library for parsing XML documents"
 TERMUX_PKG_LICENSE="MIT"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION="2.14.4"
+TERMUX_PKG_VERSION="2.14.5"
 TERMUX_PKG_SRCURL=https://download.gnome.org/sources/libxml2/${TERMUX_PKG_VERSION%.*}/libxml2-${TERMUX_PKG_VERSION}.tar.xz
-TERMUX_PKG_SHA256=24175ec30a97cfa86bdf9befb7ccf4613f8f4b2713c5103e0dd0bc9c711a2773
+TERMUX_PKG_SHA256=03d006f3537616833c16c53addcdc32a0eb20e55443cba4038307e3fa7d8d44b
 TERMUX_PKG_AUTO_UPDATE=true
 TERMUX_PKG_SETUP_PYTHON=true
 TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
