diff --git a/packages/python-pip/build.sh b/packages/python-pip/build.sh
index 347e89a..ed6aa02 100644
--- a/packages/python-pip/build.sh
+++ b/packages/python-pip/build.sh
@@ -9,7 +9,7 @@ TERMUX_PKG_SHA256=b25615caeef66bb5049a44e6df257ebae2fcc9dc774d5369a338db29b7f53f
 TERMUX_PKG_AUTO_UPDATE=true
 TERMUX_PKG_UPDATE_TAG_TYPE="newest-tag"
 TERMUX_PKG_UPDATE_VERSION_REGEXP='^\d+\.\d+(\.\d+)?$'
-TERMUX_PKG_DEPENDS="clang, make, pkg-config, python (>= 3.11.1-1)"
+TERMUX_PKG_DEPENDS="python (>= 3.11.1-1)"
 TERMUX_PKG_ANTI_BUILD_DEPENDS="clang"
 TERMUX_PKG_BREAKS="python (<< 3.11.1-1)"
 TERMUX_PKG_PLATFORM_INDEPENDENT=true
