--- a/packages/valac/build.sh
+++ b/packages/valac/build.sh
@@ -7,7 +7,7 @@ TERMUX_PKG_REVISION=1
 TERMUX_PKG_SRCURL=https://download.gnome.org/sources/vala/${TERMUX_PKG_VERSION%.*}/vala-$TERMUX_PKG_VERSION.tar.xz
 TERMUX_PKG_SHA256=f2affe7d40ab63db8e7b9ecc3f6bdc9c2fc7e3134c84ff2d795f482fe926a382
 TERMUX_PKG_AUTO_UPDATE=true
-TERMUX_PKG_DEPENDS="glib"
+TERMUX_PKG_DEPENDS="glib, gobject-introspection, graphviz"
 TERMUX_PKG_RECOMMENDS="clang, pkg-config"
 TERMUX_PKG_BREAKS="valac-dev"
 TERMUX_PKG_REPLACES="valac-dev"
