--- a/scripts/run-docker.sh
+++ b/scripts/run-docker.sh
@@ -3,7 +3,7 @@ set -e -u
 
 TERMUX_SCRIPTDIR=$(cd "$(realpath "$(dirname "$0")")"; cd ..; pwd)
 : ${TERMUX_BUILDER_IMAGE_NAME:=ghcr.io/termux/package-builder}
-: ${CONTAINER_NAME:=termux-package-builder}
+: ${CONTAINER_NAME:=termux-generator-package-builder}
 : ${TERMUX_DOCKER_RUN_EXTRA_ARGS:=}
 : ${TERMUX_DOCKER_EXEC_EXTRA_ARGS:=}
 BUILDSCRIPT_NAME=build-package.sh
@@ -131,6 +131,7 @@ APPARMOR_PARSER=""
 if command -v apparmor_parser > /dev/null; then
 	APPARMOR_PARSER="apparmor_parser"
 fi
+APPARMOR_PARSER=""
 
 if [ -z "$APPARMOR_PARSER" ] || ! $SUDO aa-status --enabled; then
 	echo "WARNING: apparmor_parser not found, AppArmor profiles will not be loaded!"
