--- a/packages/termux-tools/build.sh
+++ b/packages/termux-tools/build.sh
@@ -21,6 +21,14 @@ TERMUX_PKG_DEPENDS="bzip2, coreutils, curl, dash, diffutils, findutils, gawk, gr
 TERMUX_PKG_RECOMMENDS="ed, dos2unix, inetutils, net-tools, patch, unzip"
 
 termux_step_pre_configure() {
+	find . -type f -print0 | xargs -0 sed -i "s/com.termux/$TERMUX_APP__PACKAGE_NAME/g"
+	echo "Welcome to $TERMUX_APP__PACKAGE_NAME!
+This is a fork of \"F-Droid\" Termux!
+Do not report issues with this build upstream!
+========= Donate Here ===========
+https://termux.dev/en/donate.html
+https://opencollective.com/termux
+=================================" > motds/motd
 	autoreconf -vfi
 }
 
