--- a/packages/termux-tools/build.sh
+++ b/packages/termux-tools/build.sh
@@ -21,6 +21,15 @@ TERMUX_PKG_DEPENDS="bzip2, coreutils, curl, dash, diffutils, findutils, gawk, gr
 TERMUX_PKG_RECOMMENDS="ed, dos2unix, inetutils, net-tools, patch, unzip"
 
 termux_step_pre_configure() {
+	. "$TERMUX_SCRIPTDIR/scripts/termux_generator_utils.sh"
+   replace_termux_name "$TERMUX_PKG_SRCDIR" "$TERMUX_APP__PACKAGE_NAME"
+	echo "Welcome to $TERMUX_APP__PACKAGE_NAME!
+This is a fork of \"F-Droid\" Termux!
+Do not report issues with this build upstream!
+========= Donate Here ===========
+https://termux.dev/en/donate.html
+https://opencollective.com/termux
+=================================" > motds/motd
 	autoreconf -vfi
 }
 
--- a/packages/termux-am/build.sh
+++ b/packages/termux-am/build.sh
@@ -13,6 +13,9 @@ TERMUX_PKG_CONFLICTS="termux-tools (<< 0.51)"
 _GRADLE_VERSION=8.10.2
 
 termux_step_post_get_source() {
+	. "$TERMUX_SCRIPTDIR/scripts/termux_generator_utils.sh"
+	replace_termux_name "$TERMUX_PKG_SRCDIR" "$TERMUX_APP__PACKAGE_NAME"
+	migrate_termux_folder_tree "$TERMUX_PKG_SRCDIR" "$TERMUX_APP__PACKAGE_NAME"
 	sed -i'' -E -e "s|\@TERMUX_PREFIX\@|${TERMUX_PREFIX}|g" "$TERMUX_PKG_SRCDIR/am-libexec-packaged"
 	sed -i'' -E -e "s|\@TERMUX_APP_PACKAGE\@|${TERMUX_APP_PACKAGE}|g" "$TERMUX_PKG_SRCDIR/app/src/main/java/com/x11/termux/test/termuxam/FakeContext.java"
 }
diff --git a/packages/command-not-found/name-change-helper.patch b/packages/command-not-found/name-change-helper.patch
new file mode 100644
index 0000000..0250f2e
--- /dev/null
+++ b/packages/command-not-found/name-change-helper.patch
@@ -0,0 +1,16 @@
+diff --git a/generate-db.js b/generate-db.js
+index f1ff919..46d3c9b 100755
+--- a/generate-db.js
++++ b/generate-db.js
+@@ -6,7 +6,10 @@ import { promisify } from "node:util";
+ 
+ const gunzipAsync = promisify(gunzip);
+ 
+-const { TERMUX_SCRIPTDIR, TERMUX_PREFIX, TERMUX_ARCH } = process.env;
++const { TERMUX_SCRIPTDIR, TERMUX_ARCH } = process.env;
++// command-not-found retrieves metadata from official package mirror,
++// which stores packages built with official paths.
++const TERMUX_PREFIX = "/data/data/com.termux/files/usr";
+ 
+ if (!TERMUX_SCRIPTDIR) {
+   throw new Error("TERMUX_PKG_CACHEDIR environment variable is not defined");
