--- a/packages/proot/build.sh
+++ b/packages/proot/build.sh
@@ -19,6 +19,18 @@ TERMUX_PKG_EXTRA_MAKE_ARGS="-C src"
 export PROOT_UNBUNDLE_LOADER=$TERMUX_PREFIX/libexec/proot
 
 termux_step_pre_configure() {
+	# work around https://github.com/termux/termux-packages/issues/27475
+	termux_download "https://packages.termux.dev/apt/termux-main/pool/main/p/proot/proot_${TERMUX_PKG_VERSION}-${TERMUX_PKG_REVISION}_${TERMUX_ARCH}.deb" \
+		"$TERMUX_PKG_TMPDIR/proot.deb" SKIP_CHECKSUM
+	mkdir -p "$TERMUX_PKG_TMPDIR/proot"
+	ar x "$TERMUX_PKG_TMPDIR/proot.deb" --output="$TERMUX_PKG_TMPDIR/proot"
+	tar xf "$TERMUX_PKG_TMPDIR/proot/data.tar.xz" -C "$TERMUX_PKG_TMPDIR/proot"
+	local loader_dir="$TERMUX_PKG_TMPDIR/proot/data/data/com.termux/files/usr/libexec/proot"
+	install -Dm700 "$loader_dir/loader" -t "$PROOT_UNBUNDLE_LOADER"
+	if [[ -f "$loader_dir/loader32" ]]; then
+		install -Dm700 "$loader_dir/loader32" -t "$PROOT_UNBUNDLE_LOADER"
+	fi
+
 	CPPFLAGS+=" -DARG_MAX=131072"
 }
 
diff --git a/packages/proot/skip-building-loader.patch b/packages/proot/skip-building-loader.patch
new file mode 100644
index 0000000..7c320ec
--- /dev/null
+++ b/packages/proot/skip-building-loader.patch
@@ -0,0 +1,22 @@
+--- a/src/GNUmakefile
++++ b/src/GNUmakefile
+@@ -94,9 +94,7 @@ $(eval $(call define_from_arch.h,,HAS_POKEDATA_WORKAROUND))
+ 
+ ifdef PROOT_UNBUNDLE_LOADER
+   CFLAGS += "-DPROOT_UNBUNDLE_LOADER=\"$(PROOT_UNBUNDLE_LOADER)\""
+-proot: | loader/loader
+ ifdef HAS_LOADER_32BIT
+-proot: | loader/loader-m32
+ endif
+ else
+   OBJECTS += loader/loader-wrapped.o
+@@ -260,9 +258,7 @@ clean distclean:
+ install: proot
+ 	$($(quiet)INSTALL) -D $< $(DESTDIR)$(BINDIR)/$<
+ ifdef PROOT_UNBUNDLE_LOADER
+-	$(INSTALL) -D loader/loader $(PROOT_UNBUNDLE_LOADER)/loader
+ ifdef HAS_LOADER_32BIT
+-	$(INSTALL) -D loader/loader-m32 $(PROOT_UNBUNDLE_LOADER)/loader32
+ endif
+ endif
+ 
